### ISSUE TITLE
Incorrect patch produced

### ISSUE THREAD

**Issue by trvrm** [stefankoegl/python-json-patch#55]
``` python
src =  [{'a': 1},{'b': 2}]
dst =  [{'a': 1,'b': 2}]

patch = jsonpatch.make_patch(src,dst)
print(patch.patch)
print(patch.apply(src))
assert patch.apply(src)==dst
```

produces the following patch

``` python
[
    {'op': 'remove', 'path': '/1'}, 
    {'op': 'replace', 'value': 1, 'path': '/0/a'}
]
```

Which when applied to `src` produces 

``` python
[{'a': 1}]
```

which is obviously incorrect, and fails the assertion.

**Comment by mrhadenought** [comment]
The problem comes from the https://github.com/stefankoegl/python-json-patch/blob/master/jsonpatch.py#L772 
It should not make a patch of the operations "remove" and "add" cause it's not a source and destination objects, so need to treat differently.

**Comment by stefankoegl** [comment]
@mrhadenought thanks for your comment. As you can probably see, I didn't have enough time to devote to the development of ``jsonpatch`` recently. If you could write up a pull request, I'd make sure to verify it and release a new version promptly.

**Comment by stefankoegl** [comment]
This should have been resolved by #59. I will release a new version soon.
